Modify kube-flannel.yaml to use rbac.authorization.k8s.io/v1

vanou opened the pull request:
## Description
<!-- A few sentences describing the overall goals of the pull request's commits. 
Please include 
- the type of fix - (e.g. bug fix, new feature, documentation)
- some details on _why_ this PR should be merged
- the details of the testing you've done on it (both manual and automated)
- which components are affected by this PR
-->
**[Type of fix] :** Bug

This commit modifies `Document/kube-flannel.yaml` to use `ClusterRole` & `ClusterRoleBinding` of `rbac.authorization.k8s.io/v1`.

Before this commit, `rbac.authorization.k8s.io/v1beta1` is used at `kube-flannel.yaml`. However, from Kubernetes v1.17, this v1beta1 version is [deprecated](https://v1-17.docs.kubernetes.io/docs/setup/release/notes/#deprecations-and-removals) and `rbac.authorization.k8s.io/v1` becomes GA.

At the time of writing this commit, latest Kubernets version is v1.19. So, according to Kubernetes support policy, only v1.19, v1.18 & v1.17 are supported and it's safe to use `rbac.authorization.k8s.io/v1` now.

## Todos
- [x] Tests
- [ ] Documentation
- [ ] Release note

## Release Note
<!-- Writing a release note:
- By default, no release note action is required.
- If you're unsure whether or not your PR needs a note, ask your reviewer for guidance.
- If this PR requires a release note, update the block below to include a concise note describing
  the change and any important impacts this PR may have.
-->

```release-note
None required
```

rajatchopra reviewed (approved):
/lgtm

AleksandrNull reviewed (commented):
Looks good. Can you please also change docs to reflect that change (just to have a reference that new roles wouldn't be compatible with <1.17 ?

vanou commented:
Thanks @rajatchopra & @AleksandrNull !
I modified `README.md` & `Documentation/kubernetes.md`.

I split `Older versions of Kubernetes` section of `Documentation/kubernetes.md` into two subsections:

1. For Kubernetes v1.6~v1.15

   This subsection holds all text `Older versions of Kubernetes` section held before this commit and talks about how to use manifests under `Documentation/k8s-manifests`.

   I name this subsection `v1.6~v1.15` because

   * `rbac.authorization.k8s.io/v1beta1` used in `Documentation/k8s-manifests/kube-flannel-rbac.yml` appears from Kubernetes version v1.6 ([In v1.5](https://github.com/kubernetes/kubernetes/blob/release-1.5/api/openapi-spec/swagger.json), no v1beta1 and only v1alpha1. But from [v1.6](https://github.com/kubernetes/kubernetes/blob/release-1.6/api/openapi-spec/swagger.json), v1beta1 appears.)

   * `extensions/v1beta1 DaemonSet` used in `Documentation/k8s-manifests/kube-flannel-legacy.yml` is [deprecated in k8s v1.16](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/)

   * another k8s resource version is supported in this range (v1.6~v1.15)

2. For Kubernetes v1.16

   This subsection refers that user should replace `rbac.authorization.k8s.io/v1` in `kube-flannel.yaml` to `rbac.authorization.k8s.io/v1beta1` under K8s v1.16.

vanou commented:
Kindly ping:
Hello @rajatchopra & @AleksandrNull . I know you're very busy. It's thankful for you to give feedback when you have a time.

Thank you.

czomo commented on Documentation/kubernetes.md:
```diff
@@ -31,6 +33,10 @@ This file does not bundle RBAC permissions. If you need those, run
 
+### For Kubernetes v1.16
+
+`kube-flannel.yaml` uses `ClusterRole` & `ClusterRoleBinding` of `rbac.authorization.k8s.io/v1`. When you use Kubernetes v1.16, you should replace `rbac.authorization.k8s.io/v1` to `rbac.authorization.k8s.io/v1beta1` because `rbac.authorization.k8s.io/v1` had become GA from Kubernetes v1.17.
```
rbac.authorization.k8s.io/v1 is added to k8s since 1.8.0, we don't need to add that
